db/migrate/20190914204954_replace_assignments_and_grade_entry_forms_with_assessments.rb

```diff
@@ -0,0 +1,185 @@
+class ReplaceAssignmentsAndGradeEntryFormsWithAssessments < ActiveRecord::Migration[6.0]
+  def change
+
+    # Ensure rollback contains the correct names & indexes
+    reversible do |change|
+      change.down do
+        add_index :assignment_files, [:assignment_id, :filename], unique: true
+        add_index :checkbox_criteria, [:assignment_id, :name], unique: true
+        add_index :groupings, [:assignment_id, :group_id], name: "groupings_u1", unique: true
+        add_index :rubric_criteria, [:assignment_id, :name], name: "rubric_criteria_index_1", unique: true
+        add_index :flexible_criteria, [:assignment_id, :name], unique: true
+
+        add_index :assignments, [:short_identifier], unique: true
+        add_index :grade_entry_forms, [:short_identifier], unique: true
+
+        add_foreign_key :assignment_files, :assignments, name: "fk_assignment_files_assignments", on_delete: :cascade
+        add_foreign_key :assignment_stats, :assignments, name: "fk_assignment_stats_assignments", on_delete: :cascade
+        add_foreign_key :checkbox_criteria, :assignments
+        add_foreign_key :exam_templates, :assignments
+        add_foreign_key :groupings, :assignments, name: "fk_groupings_assignments"
+        add_foreign_key :rubric_criteria, :assignments, name: "fk_rubric_criteria_assignments", on_delete: :cascade
+        add_foreign_key :test_groups, :assignments
+      end
+    end
+
+    create_table :assessments, id: :serial, force: :cascade do |t|
+      t.string :short_identifier, null: false
+      t.string :type
+      t.string :description
+      t.text :message
+      t.date :date
```

Comment:
Keep the name `due_date` rather than `date` (i.e., I prefer the assignment name rather than the grade entry form name).